kotest-assertions/kotest-assertions-json/src/commonMain/kotlin/io/kotest/assertions/json/schema/matchSchema.kt

```diff
@@ -77,13 +77,17 @@ private fun validate(
    fun violation(message: String) =
       listOf(SchemaViolation(currentPath, message))
 
+   fun violation(conditionResult: Boolean, message: String) = if(conditionResult) violation(message) else emptyList()
```

Comment:
It was not immediately clear to me that this was a different `violation` function than the other invocations. I think we should either rename this one, or just inline the conditional
```suggestion
fun violationIf(condition: Boolean, messageFn: () -> String) = if(conditionResult) violation(message) else emptyList()
   
val sizeViolation = violationIf(tree.elements.size < expected.minItems || tree.elements.size > expected.maxItems) {
   "Expected items between ${expected.minItems} and ${expected.maxItems}, but was ${tree.elements.size}"
}
```

or:

```suggestion
val sizeViolation = if (tree.elements.size < expected.minItems || tree.elements.size > expected.maxItems) violation("Expected items between ${expected.minItems} and ${expected.maxItems}, but was ${tree.elements.size}") else emptyList()
```

Reply:
@Kantis reason why I did it is because when we deal with more violations (in future) then we will have huge if block `if ... else emptyList` so I tried to do something with it. Another thought was to implement array violations in separate file. 

But your idea to rename this is totally fine !

Reply:
Ah.. I understand and agree that it would be a good idea to break it down a bit somewhere down the line :)